### DISCUSSION TITLE
Should Ctrl+C be propagated as a signal instead of captured and returned?

### DISCUSSION THREAD

**Discussion by mikaelmello** [mikaelmello/inquire#201] [Ideas]
Inquire captures Ctrl+C, maybe it should propagate

**Comment by IniterWorker** [comment]
Should we allow the user to decide?

**Comment by mikaelmello** [comment]
Certainly. I'm thinking though that it should propagate by default (a breaking change), and then if they want to handle it themselves, they can opt-in. Wdyt?

For context, I'm really aiming at a v1 now so revamps/breaking changes of the sort should be acceptable. It can be a whole thing like "inquire is stable, see what changes"

**Reply by crabdancing** [reply]
Propagate by default seems totally reasonable. Besides, it would really only be a breaking change on systems that already do not follow the Ctrl+C standard. I can't imagine this being a problem unless someone has done something very strange, like scripting the interactive inputs to a Rust program made with `inquire`.

**Comment by crabdancing** [comment]
Ctrl+C isn't even universal on all terminal configurations. Would really like a way to make my program not close when hitting it.
